### PR TITLE
Tasks/user doc enhancements

### PR DESCRIPTION
**Pull request by rhornung67**
# Summary

- This PR contains some user doc cleanup and enhancements based on comments made by various people. This is the last user guide doc PR before the v2022.10.0 release.

The RTD generated docs for this PR are here: https://raja.readthedocs.io/en/tasks-user-doc-enhancements/

### CODE REVIEW

File: docs/sphinx/user_guide/getting_started.rst [hunk 1]
```diff
@@ -13,17 +13,19 @@
...
 ============
 
-The primary requirement for using RAJA is a C++14 compliant compiler.
-Accessing various programming model back-ends requires that they be supported
-by the compiler you chose. Available options and how to enable or disable 
-them are described in :ref:`configopt-label`. To build RAJA in its most basic
-form and use its simplest features:
+The primary requirement for using RAJA is a C++14 standard compliant compiler.
+Using features, such as various programming model back-ends like CUDA or HIP, 
+requires that they be supported by the compiler you chose. Available RAJA
```
Comment (MrBurmark) [inline docs/sphinx/user_guide/getting_started.rst:24]:
```suggestion
Certain features, such as various programming model back-ends like CUDA or HIP, 
must be supported by the compiler you chose. Available RAJA
```

File: docs/sphinx/user_guide/getting_started.rst [hunk 2]
```diff
@@ -187,48 +222,55 @@ When using the NVIDIA nvcc compiler for RAJA CUDA functionality, the variables:
...
-``-DRAJA_ENABLE_EXTERNAL_CUB=On -DCUB_DIR=<path/to/cub>``.
-
-.. note:: It is important to note that the CUDA toolkit version of cub is
+            CMake will report this as an error and stop processing.
+
+Also, RAJA relies on the CUB CUDA utilities library, mentioned earlier, for 
+some CUDA back-end functionality. The CUB version included in the CUDA toolkit 
+installation is used by default when available. This is the case for CUDA 
+version 11 and later. RAJA includes a CUB submodule that can be used with 
+older versions of CUDA. To use an external CUB installation, provide the
```
Comment (MrBurmark) [inline docs/sphinx/user_guide/getting_started.rst:257]:
```suggestion
version 11 and later. RAJA includes a CUB submodule that is used by default with 
older versions of CUDA. To use an external CUB installation, provide the 
```

File: docs/sphinx/user_guide/getting_started.rst [hunk 3]
```diff
@@ -187,48 +222,55 @@ When using the NVIDIA nvcc compiler for RAJA CUDA functionality, the variables:
...
+
+.. note:: The CUDA toolkit version of CUB is
           required for compatibility with the CUDA toolkit version of thrust
-          starting with CUDA toolkit version v11.0.0. So, if you build
-          RAJA with CUDA version 11 or higher you should use the version of
+          starting with CUDA version 11.0.0. So, if you build
+          RAJA with CUDA version 11 or higher, you should use the version of
           CUB contained in the CUDA toolkit version you are using to use 
-          Thrust and be compatible with libraries that use Thrust.
+          Thrust and to be compatible with libraries that use Thrust.
```
Comment (MrBurmark) [inline docs/sphinx/user_guide/getting_started.rst:271]: awkward

File: docs/sphinx/user_guide/getting_started.rst (hunk 2 above)
Comment (rhornung67) [reply docs/sphinx/user_guide/getting_started.rst:257]: fixed

File: docs/sphinx/user_guide/getting_started.rst (hunk 3 above)
Comment (rhornung67) [reply docs/sphinx/user_guide/getting_started.rst:271]: fixed

**Review by MrBurmark — APPROVED** [review]

**Pull request merged**